contracts/protocol/modules/SnapshotDelegationModule.sol

```diff
@@ -0,0 +1,45 @@
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+import { IController } from "../../interfaces/IController.sol";
+import { ISetToken } from "../../interfaces/ISetToken.sol";
+import { ModuleBase } from "../lib/ModuleBase.sol";
+import { DelegateRegistry } from "../../../external/contracts/snapshot/DelegateRegistry.sol";
+
+
+contract SnapshotDelegationModule is ModuleBase {
+
+    event Delegated(bytes32 _id, address _to);
+
+    DelegateRegistry delegateRegistry;
+
+    constructor(
+        IController _controller, 
+        DelegateRegistry _delegateRegistry
+    ) 
+        public
+        ModuleBase(_controller) 
+    {
+        delegateRegistry = _delegateRegistry;
+    }
+
+    function initialize(
+        ISetToken _setToken
+    )
+        external
+        onlyValidAndPendingSet(_setToken)
+        onlySetManager(_setToken, msg.sender)
+    {
+        _setToken.initializeModule();
+    }
+
+    function removeModule() external override {}
+
+    function delegate(ISetToken _setToken, bytes32 _id, address _to) external onlyManagerAndValidSet(_setToken) {
```

Comment:
At first glance, it seems like we can potentially implement this as a governance adapter rather than a separate module. Do we require the _id parameter in the snapshot delegate function? Or can it be just a constant which allows us to implement using the existing governance adapter interface

Reply:
The id parameter is used to denote what snapshot space you are delegating for. If you pass in 0, then it delegates governance for all spaces to the supplied address. If you were to pass in "balancer", then it would only delegate the contracts BAL voting power to the supplied address. We could have this id parameter just always be 0, but we would be losing out on the ability to selectively delegate different snapshot spaces to different addresses.

Reply:
Cool, makes sense. Does seem like the most likely scenario is DPI (or any other Set) delegating all its snapshot voting power at once? Since contracts can't sign messages. Would make a lot more sense if you're an EOA and wanting to delegate votes on specific projects

Reply:
Definitely agree that delegating everything to one address is way more likely. Should I change it to a `GovernanceModule` adapter then?

Reply:
Yeah let's do an adapter then. Should be straightforward implementing the getDelegateCalldata interface. We could even implement getRevokeCalldata for clearDelegate

Reply:
Sounds good. Working on it right now. One thing I noticed is that the adapter tests just ensure that the returned values of each getCalldata function are equal to the expected values rather than doing a full end-to-end test of the adapter being called by the module. Should I follow this convention?

Reply:
Yeah these integration tests are in another folder. E.g Compound governance adapter here https://github.com/SetProtocol/set-protocol-v2-contracts/blob/master/test/integration/compoundGovernanceModule.spec.ts